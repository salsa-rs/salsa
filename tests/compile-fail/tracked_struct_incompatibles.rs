#[salsa::tracked(returns(ref))]
struct TrackedWithRetRef {
    field: u32,
}

#[salsa::tracked(specify)]
struct TrackedSructWithSpecify {
    field: u32,
}

#[salsa::tracked(no_eq)]
struct TrackedStructWithNoEq {
    field: u32,
}

#[salsa::tracked(db = Db)]
struct TrackedStructWithDb {
    field: u32,
}

#[salsa::tracked(recover_fn = recover)]
struct TrackedStructWithRecover {
    field: u32,
}

#[salsa::tracked(lru = 12)]
struct TrackedStructWithLru {
    field: u32,
}

#[salsa::tracked(revisions = 12)]
struct TrackedStructWithRevisions {
    field: u32,
}

fn main() {}
