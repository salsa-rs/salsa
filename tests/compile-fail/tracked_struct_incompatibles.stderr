error: `returns` option not allowed here
 --> tests/compile-fail/tracked_struct_incompatibles.rs:1:18
  |
1 | #[salsa::tracked(returns(ref))]
  |                  ^^^^^^^

error: `specify` option not allowed here
 --> tests/compile-fail/tracked_struct_incompatibles.rs:6:18
  |
6 | #[salsa::tracked(specify)]
  |                  ^^^^^^^

error: `no_eq` option not allowed here
  --> tests/compile-fail/tracked_struct_incompatibles.rs:11:18
   |
11 | #[salsa::tracked(no_eq)]
   |                  ^^^^^

error: `db` option not allowed here
  --> tests/compile-fail/tracked_struct_incompatibles.rs:16:18
   |
16 | #[salsa::tracked(db = Db)]
   |                  ^^

error: unrecognized option `recover_fn`
  --> tests/compile-fail/tracked_struct_incompatibles.rs:21:18
   |
21 | #[salsa::tracked(recover_fn = recover)]
   |                  ^^^^^^^^^^

error: `lru` option not allowed here
  --> tests/compile-fail/tracked_struct_incompatibles.rs:26:18
   |
26 | #[salsa::tracked(lru = 12)]
   |                  ^^^

error: `revisions` option not allowed here
  --> tests/compile-fail/tracked_struct_incompatibles.rs:31:18
   |
31 | #[salsa::tracked(revisions = 12)]
   |                  ^^^^^^^^^
